### PR TITLE
Added "Bargonia" bar sign!

### PR DESCRIPTION
**Pull request by DrAmazing343**
## About The Pull Request
I made this submission for the Bar Sign contest, and while I didn't make it into any of the higher accolades, I'd still love to see it make it into the game!

## Why It's Good For The Game
Cargo's always had fun making all sorts of structures in the warehouse, and the fabled Cargo Bar is one of these that seems to be a fan-favorite. I set out to make a sign to be a nice nod to these bars, and for something else for people to put up on the wall to decorate the bars!
bargoniaupscaledx4

## Changelog

:cl: Jane
add: Bargonia Bar Sign for Cargo Bar Enjoyers
/:cl:

### CODE REVIEW

**Comment by BlueMemesauce** [comment]
It feels more like a bar nation name rather than a cargo reference

File: code/game/machinery/barsigns.dm
Comment (san7890) [inline code/game/machinery/barsigns.dm:399]:
```suggestion
    icon_state = "bargonia"
```

your code won't compile otherwise

**Review by san7890 — CHANGES_REQUESTED** [review]

**Comment by DrAmazing343** [comment]
> It feels more like a bar nation name rather than a cargo reference

We call Cargo that's "seceded" from the station Cargonia. Bargonia is a reference to the term Cargonia.

File: code/game/machinery/barsigns.dm
Comment (DrAmazing343) [reply code/game/machinery/barsigns.dm]: Committed the suggestion. Thank you!

**Comment by deathrobotpunch** [comment]
> > It feels more like a bar nation name rather than a cargo reference
> 
> We call Cargo that's "seceded" from the station Cargonia. Bargonia is a reference to the term Cargonia.

*Bargo

**Comment by Loafin34** [comment]
why a mothroach? why not a mulebot?

**Comment by DrAmazing343** [comment]
> why a mothroach? why not a mulebot?

Mulebot would be fun, but I really liked the visual of cracking open a wooden crate and mothroaches come in the wooden critter containers!

**Review by ZephyrTFA — CHANGES_REQUESTED** [review]
We just got a Mothroach themed sign, I'm not inserted in this one also having one at the moment

**Comment by DrAmazing343** [comment]
> We just got a Mothroach themed sign, I'm not inserted in this one also having one at the moment

Is there a level of quality issue with my sign? Any constructive criticism aside from a disagreement in theming? Something concrete that I can improve on?

**Comment by ZephyrTFA** [comment]
> > We just got a Mothroach themed sign, I'm not inserted in this one also having one at the moment
> 
> Is there a level of quality issue with my sign? Any constructive criticism aside from a disagreement in theming? Something concrete that I can improve on?

Don't get me wrong. The barsign is great. I just don't want another mothroach one so soon after we just had Maltroach merged. I feel like it detracts from the novelty of the mothroach sign and siphons attention away from it for the wrong reason.
This is in no way a blocking review and just an opinion from me for other maintainers

**Comment by DrAmazing343** [comment]
> > > We just got a Mothroach themed sign, I'm not inserted in this one also having one at the moment
> > 
> > 
> > Is there a level of quality issue with my sign? Any constructive criticism aside from a disagreement in theming? Something concrete that I can improve on?
> 
> Don't get me wrong. The barsign is great. I just don't want another mothroach one so soon after we just had Maltroach merged. I feel like it detracts from the novelty of the mothroach sign and siphons attention away from it for the wrong reason. This is in no way a blocking review and just an opinion from me for other maintainers

Ah, okay, I don't know how all the PR business works, so I was really worried for a moment. I understand your concern, but having spoken to the Maltroach's creator, they quite liked my sign as well! This was my best foot forward to create a nice novel sign for one of my favorite departments/gimmicks; I hope more people don't take it as an attempt to steal thunder from the very very impressive sign that Moff Nyan created.  

If anyone reviewing/looking at the PR has any creative suggestions, I'd love to hear them out as well, since this is my first addition to the codebase and I'd like to have it be the best I can make it if it's gonna make it in at all.

**Review by ZephyrTFA — APPROVED** [review]

**Comment by Wallemations** [comment]
I'd recommend making the letters more 3D, or perhaps even having them be dimmed out before they're lit up rather than simply appearing, more like neon lights rather than just a couple pixels on a purple background?

**Comment by DrAmazing343** [comment]
> I'd recommend making the letters more 3D, or perhaps even having them be dimmed out before they're lit up rather than simply appearing, more like neon lights rather than just a couple pixels on a purple background?

Right, please check out the new .dmi! I've also updated the PR page's gif to be upscaled to SEE IT, but that's the aseprite raw that doesn't have the .dmi's better timings, so it's mostly just to show off the updated 3D/neon tube'd up sprite.

**Comment by Wallemations** [comment]
> > I'd recommend making the letters more 3D, or perhaps even having them be dimmed out before they're lit up rather than simply appearing, more like neon lights rather than just a couple pixels on a purple background?
> 
> Right, please check out the new .dmi! I've also updated the PR page's gif to be upscaled to SEE IT, but that's the aseprite raw that doesn't have the .dmi's better timings, so it's mostly just to show off the updated 3D/neon tube'd up sprite.

Big improvement, though I'll ask if you've tried making the shadow on the top of the letters instead to make them seem like they're popping out from an above view?
No need to resprite the whole thing to check, just the final frame edited to see how it would look should be fine

**Comment by DrAmazing343** [comment]
> > > I'd recommend making the letters more 3D, or perhaps even having them be dimmed out before they're lit up rather than simply appearing, more like neon lights rather than just a couple pixels on a purple background?
> > 
> > 
> > Right, please check out the new .dmi! I've also updated the PR page's gif to be upscaled to SEE IT, but that's the aseprite raw that doesn't have the .dmi's better timings, so it's mostly just to show off the updated 3D/neon tube'd up sprite.
> 
> Big improvement, though I'll ask if you've tried making the shadow on the top of the letters instead to make them seem like they're popping out from an above view? No need to resprite the whole thing to check, just the final frame edited to see how it would look should be fine

Do you mean adding the shadow-shading line to the tops of the lettering as opposed to the left-side? I don't know exactly what area you're talking about here, but I'll try to resprite what I think you're talking about in the morning. If you come back to review in the meantime, a bit of clarification would go a long ways to helping me.

**Comment by DrAmazing343** [comment]
bargonia top shading test

Right, trying this one on for size. I think I personally liked the side-lighting better, but if this is what you meant to try and you think it looks better, I'll redo the .dmi/lightmask for it!

**Comment by Wallemations** [comment]
> bargonia top shading test
> 
> 
> 
> Right, trying this one on for size. I think I personally liked the side-lighting better, but if this is what you meant to try and you think it looks better, I'll redo the .dmi/lightmask for it!

I think I personally prefer this one, yea
Remember to add the darker colors for the inside of the letters too

**Comment by DrAmazing343** [comment]
> I think I personally prefer this one, yea Remember to add the darker colors for the inside of the letters too

Adjusted and committed! Any other issues I need to work out?

**Comment by Wallemations** [comment]
the merge conflicts, mostly.
the influx of bar signs means that every time I merge one every other barsign PR has to adjust for it

**Comment by DrAmazing343** [comment]
Ignore most of the fuckery above, I'm a novice and it shows. Everything SHOULD be clear for merging, unless I've missed or messed up something?

**Review by Wallemations — APPROVED** [review]
since there's a lot goin on here, once that thing ghommie suggested gets fixed if any other maints wanna merge it before I do go ahead
Otherwise I'll check in again later and merge it myself

**Pull request merged**
